### ISSUE TITLE
Change processing

### ISSUE THREAD

**Issue by ctmay4** [javers/javers#66]
I am trying to figure out how the library would be used to show a traditional "changelog".  By traditional I am referring to something that is grouped together by "commit" then by field.

``` java
        List<Change> changes = javers.getChangeHistory(InstanceIdDTO.instanceId(table.getInternalId(), Entity.class), 5);

        for (Change change : changes) {
            CommitMetadata commit = change.getCommitMetadata().get();
            System.out.print(commit.getId() + ":" + commit.getCommitDate() + ":" + commit.getAuthor() + " -- ");

            // What type of change is it?

            // I could say "if (change instanceof ValueChange)" and check for every type
        }

```

So a couple of questions:
1.  The documentation is missing in some places, so I may have missed it.  Is there a way to get "commits" instead of "changes"?  Commits would store the CommitMetaData and have a list of Changes?  Otherwise, I would just build it myself, which is not too difficult.
2.  When I am looping over the changes, how do I know what type of change it is?  I.E NewObject, ValueChange, MapChange, etc.  In all the examples I saw, you were retrieving changes by type.  However I would like to build a complete changelog so I don't know beforehand which types to ask for.  I could do a bunch of casting based on instanceof but that doesn't see right.  Is there a more elegant way, given a Change, to determine its actual type?

**Comment by bartoszwalacik** [comment]
https://github.com/javers/javers/issues/73 is developed and put to pull request, stay tuned

**Comment by bartoszwalacik** [comment]
https://github.com/javers/javers/issues/73 Listing newObject event on change history 
is released in 1.0.1

**Comment by ctmay4** [comment]
Just tested this and it is working correctly for my tests.

```
commit 3.0, author:me@here.com, Jan 6, 2015 2:11:08 PM
  changed object: entities.Table/54ac334c102cc3112b170092
    value changed on 'notes' property: 'Some notes' -> 'Changed notes'
    value changed on 'footnotes' property: 'null' -> 'A footnote'
commit 2.0, author:me@here.com, Jan 6, 2015 2:11:08 PM
  changed object: entities.Table/54ac334c102cc3112b170092
    value changed on 'notes' property: 'null' -> 'Some notes'
commit 1.0, author:me@here.com, Jan 6, 2015 2:11:08 PM
    new object: 'entities.Table/54ac334c102cc3112b170092
```

One VERY minor issue in SimpleTestChangeLog.java:

```java
    @Override
    public void onNewObject(NewObject newObject) {
        appendln("    new object: '" + newObject.getAffectedGlobalId());
    }
```

There is an unmatched single quote in the string that is not in the other text outputs.

The only outstanding issue I have found is covered in #71.

**Comment by bartoszwalacik** [comment]
right, I'll remove this quote mark,
we are working on #71.

**Comment by ctmay4** [comment]
Sounds great.  Thanks again!

**Comment by bartoszwalacik** [comment]
Hi, last issue from this thread ( tracking a top-level object deletion #71) is solved
 and to be released in version 1.0.3

There is a new method added to Javers:
```
   /**
     * Marks given object as deleted.
     * <br/><br/>
     * 
     * Unlike {@link Javers#commit(String, Object)}, this method is shallow
     * and affects only given object.
     * <br/><br/>
     *
     * This method doesn't delete anything from JaVers repository.
     * It just persists 'terminal snapshot' of given object.
     *
     * @param deleted object to be marked as deleted
     */
    public Commit commitShallowDelete(String author, Object deleted) {
```

If you need early access to this new version,
download 1.0.3-SNAPSHOT from our snapshots repository
https://oss.sonatype.org/content/repositories/snapshots
